src/test/java/ru/hogwards/school/controller/FacultyControllerTest.java

```diff
@@ -0,0 +1,291 @@
+                    .andExpect(result -> {
+                        String responseString = result.getResponse().getContentAsString();
+                        assertThat(responseString).isNotNull();
+                        assertThat(responseString).isEqualTo("Факультет с id = 2 не найден!");
+                    });
+            verify(facultyRepository, never()).delete(any());
+        }
+
+        @Test
+        public void findAllTest() throws Exception {
+            List<Faculty> faculties = Stream.iterate(1, id -> id + 1)
+                    .map(this::generate)
+                    .limit(20)
+                    .collect(Collectors.toList());
+            List<FacultyDtoOut> expectedResult = faculties.stream()
+                    .map(facultyMapper::toDto)
+                    .collect(Collectors.toList());
+
+            when(facultyRepository.findAll()).thenReturn(faculties);
+
+            mockMvc.perform(
+                            MockMvcRequestBuilders.get("/faculties")
+                    ).andExpect(MockMvcResultMatchers.status().isOk())
+                    .andExpect(result -> {
+                        List<FacultyDtoOut> facultyDtoOuts = objectMapper.readValue(
+                                result.getResponse().getContentAsString(),
+                                new TypeReference<>() {
+                                }
+                        );
+                        assertThat(facultyDtoOuts)
+                                .isNotNull()
+                                .isNotEmpty();
+                        Stream.iterate(0, index -> index + 1)
+                                .limit(facultyDtoOuts.size())
+                                .forEach(index -> {
+                                    FacultyDtoOut facultyDtoOut = facultyDtoOuts.get(index);
+                                    FacultyDtoOut expected = expectedResult.get(index);
+                                    assertThat(facultyDtoOut.getId()).isEqualTo(expected.getId());
+                                    assertThat(facultyDtoOut.getColour()).isEqualTo(expected.getColour());
+                                    assertThat(facultyDtoOut.getName()).isEqualTo(expected.getName());
+                                });
+                    });
+
+            String colour = faculties.get(0).getColour();
+            faculties = faculties.stream()
+                    .filter(faculty -> faculty.getColour().equals(colour))
+                    .collect(Collectors.toList());
+            List<FacultyDtoOut> expectedResult2 = faculties.stream()
+                    .filter(faculty -> faculty.getColour().equals(colour))
+                    .map(facultyMapper::toDto)
+                    .collect(Collectors.toList());
+            when(facultyRepository.findAllByColour(eq(colour))).thenReturn(faculties);
+
+            mockMvc.perform(
+                            MockMvcRequestBuilders.get("/faculties?colour={colour}", colour)
```

Comment:
```suggestion
                            MockMvcRequestBuilders.get("/faculties?color={colour}", colour)
```

Передаётся не тот параметр